lib/git-hooks.js

```diff
@@ -123,23 +125,22 @@ module.exports = {
  *
  * @param {String[]} hooks List of hook names to execute.
  * @param {String[]} args
- * @param {vow.Deferred} [defer]
+ * @param {Number} Status code.
  */
-function runHooks(hooks, args, defer) {
-    defer = defer || vow.defer();
+function runHooks(hooks, args) {
+    var status;
+
     if (!hooks.length) {
-        return defer.resolve(0);
+        return consts.STATUS.SUCCESS;
     }
 
-    var hook = spawn(hooks.shift(), args, {stdio: 'inherit'});
-    hook.on('close', function (code) {
-        if (code === 0) {
-            runHooks(hooks, args, defer);
-        } else {
-            defer.reject(code);
-        }
-    });
-    return defer.promise();
+    var hook = spawnSync(hooks.shift(), args, {stdio: 'inherit'});
```

Comment:
sync spawn is bad. You'll black screen during hook's execution + you must specify enormous buffer size because you don't how much output hooks will produce.

So run method should be async.